src/type-exercises.ts

```diff
@@ -0,0 +1,262 @@
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, '')
+);
+
+console.log('Ommiting boolean type from the object');
+console.log(
+    greeting<customType, boolean>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, false)
+);
+
+console.log('Ommiting number type from the object');
+console.log(
+    greeting<customType, number>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, 0)
+);
+
+/**
+ * Exercise #2: Implement the utility type `If<C, T, F>`, which evaluates a condition `C`
+ * and returns one of two possible types:
+ * - `T` if `C` is `true`
+ * - `F` if `C` is `false`
+ *
+ * @description
+ * - `C` is expected to be either `true` or `false`.
+ * - `T` and `F` can be any type.
+ *
+ * @example
+ * type A = If<true, 'a', 'b'>;  // expected to be 'a'
+ * type B = If<false, 'a', 'b'>; // expected to be 'b'
+ */
+
+// Add here your solution
+type If<C extends boolean, T, F> = C extends true ? T : F;
+
+// Add here your example
+console.log('Exercise #2: Implement the utility type `If<C, T, F>` 👀');
+console.log('C is true');
+type A = If<true, 'a', 1>;
+console.log('Type A:', 'a' as A);
+
+console.log('C is false');
+type B = If<false, 'a', 1>;
+console.log('Type B:', 2 as B);
+
+/**
+ * Exercise #3: Recreate the built-in `Readonly<T>` utility type without using it.
+ *
+ * @description
+ * Constructs a type that makes all properties of `T` readonly.
+ * This means the properties of the resulting type cannot be reassigned.
+ *
+ * @example
+ * interface Todo {
+ *   title: string;
+ *   description: string;
+ * }
+ *
+ * const todo: MyReadonly<Todo> = {
+ *   title: "Hey",
+ *   description: "foobar"
+ * };
+ *
+ * todo.title = "Hello";       // Error: cannot reassign a readonly property
+ * todo.description = "barFoo"; // Error: cannot reassign a readonly property
+ */
+
+
+// Add here your solution
+type MyReadonly<T> = {
+    readonly [K in keyof T]: T[K];
+};
+
+interface Todo {
+    title: string;
+    description: string;
+}
+
+const todo: MyReadonly<Todo> = {
+    title: "Hey",
+    description: "foobar"
+};
+
+// Add here your example
+console.log('Exercise #3: Recreate the built-in `Readonly<T>` utility type without using it ☃️');
+// todo.title = 'Hello'; 
+console.log(todo.title);
+
+
+/**
+ * Exercise #4: Recreate the built-in `ReturnType<T>` utility type without using it.
+ *
+ * @description
+ * The `MyReturnType<T>` utility type extracts the return type of a function type `T`.
+ *
+ * @example
+ * const fn = (v: boolean) => {
+ *   if (v) {
+ *     return 1;
+ *   } else {
+ *     return 2;
+ *   }
+ * };
+ *
+ * type a = MyReturnType<typeof fn>; // expected to be "1 | 2"
+ */
+
+// Add here your solution
+// type If<C extends boolean, T, F> = C extends true ? T : F;
+type MyReturnType<T> = T extends (...args: any[]) => infer R ? R : any;
```

Comment:
Why do we need the `(...args: any[])` ? Can it be done using `()` instead?
Explain the differences

Reply:
We need `(...args: any[])` because it allows the type to work with functions regardless of whether they have parameters or not. This syntax ensures that the conditional type checks for any function, whether it accepts arguments or not.

If we replace `(...args: any[])` with `()`, the type will only work with functions that explicitly have no parameters.